### PR TITLE
Ensure that unsafe_view doesn't check bounds

### PR DESCRIPTION
**Pull request by timholy**

### CODE REVIEW

File: base/subarray.jl [hunk 1]
```diff
@@ -112,7 +112,7 @@ _maybe_reindex{C<:AbstractCartesianIndex{1}}(V, I, A::Tuple{AbstractArray{C}, Va
 _maybe_reindex(V, I, A::Tuple{Any, Vararg{Any}}) = (@_inline_meta; _maybe_reindex(V, I, tail(A)))
 function _maybe_reindex(V, I, ::Tuple{})
     @_inline_meta
-    idxs = to_indices(V.parent, reindex(V, V.indexes, I))
+    @inbounds idxs = to_indices(V.parent, reindex(V, V.indexes, I))
```
Comment (mbauman) [inline base/subarray.jl]: :+1:

File: test/subarray.jl [hunk 2]
```diff
@@ -540,3 +540,10 @@ let
     @test Base.linearindexing(view(a, :, :)) == Base.LinearFast()
     @test isbits(view(a, :, :))
 end
+
+# Ensure unsafe_view doesn't check bounds
+let A = rand(10,10)
+    B = view(A, 4:7, 4:7)
+    C = Base.unsafe_view(B, -2:7, -2:7)
+    @test C == A
```
Comment (mbauman) [inline test/subarray.jl]: I'm afraid this won't work because we run tests with `--checkbounds=yes`.

Comment (KristofferC) [inline test/subarray.jl]: Maybe could be put it in `boundscheck_exec.jl`

**Comment by timholy** [comment]
Thanks, @KristofferC, that seems to have done the trick.

**Review by mbauman — APPROVED** [review]

File: base/subarray.jl [hunk 3]
```diff
@@ -112,7 +112,7 @@ _maybe_reindex{C<:AbstractCartesianIndex{1}}(V, I, A::Tuple{AbstractArray{C}, Va
...
     SubArray(V.parent, idxs)
```
Comment (tkelman) [inline base/subarray.jl]: on release-0.5, these lines look like
```julia
    idxs = reindex(V, V.indexes, to_indexes(I...))
    SubArray(V.parent, idxs, map(unsafe_length, (index_shape(V.parent, idxs...))))
```

is the simple `@inbounds` addition to the first line equally valid there?

Comment (timholy) [inline base/subarray.jl]:
```diff
$ git diff
diff --git a/base/subarray.jl b/base/subarray.jl
index a675fd7..75ca6fd 100644
--- a/base/subarray.jl
+++ b/base/subarray.jl
@@ -82,7 +82,7 @@ function unsafe_view{T,N}(A::AbstractArray{T,N}, I::Vararg{ViewIndex,N})
 end
 function unsafe_view{T,N}(V::SubArray{T,N}, I::Vararg{ViewIndex,N})
     @_inline_meta
-    idxs = reindex(V, V.indexes, to_indexes(I...))
+    @inbounds idxs = reindex(V, V.indexes, to_indexes(I...))
     SubArray(V.parent, idxs, map(unsafe_length, (index_shape(V.parent, idxs...))))
 end
```
(This against release-0.5.)

**Pull request merged**